### PR TITLE
Added support to use Pathlib

### PR DESCRIPTION
**Pull request by SigireddyBalasai**
for the issue #90 added the pathlib support

### CODE REVIEW

**Comment by SigireddyBalasai** [comment]
@microsoft-github-policy-service agree

File: src/markitdown/_markitdown.py [hunk 1]
```diff
@@ -13,7 +13,8 @@
 import tempfile
 import traceback
 import zipfile
-from typing import Any, Dict, List, Optional, Union
+from typing import Any, Dict, List, Optional, Union, overload
```
Comment (l-lumin) [inline src/markitdown/_markitdown.py:16]: what is the purpose of  `overload`?

File: src/markitdown/_markitdown.py [hunk 2]
```diff
@@ -1042,7 +1043,7 @@ def __init__(
         self.register_page_converter(ZipConverter())
 
     def convert(
-        self, source: Union[str, requests.Response], **kwargs: Any
+        self, source: Union[str, requests.Response, Path], **kwargs: Any
```
Comment (l-lumin) [inline src/markitdown/_markitdown.py:1046]: could you update the docstring as well?

File: tests/test_markitdown.py [hunk 3]
```diff
@@ -229,8 +230,104 @@ def test_markitdown_exiftool() -> None:
...
+    reason="do not run if exiftool is not installed",
+)
+def test_markitdown_exiftool_pathlib() -> None:
+    markitdown = MarkItDown()
+
+    # Test JPG metadata processing
+    result = markitdown.convert(Path(TEST_FILES_DIR) / "test.jpg")
+    for key in JPG_TEST_EXIFTOOL:
+        target = f"{key}: {JPG_TEST_EXIFTOOL[key]}"
+        assert target in result.text_content
```
Comment (l-lumin) [inline tests/test_markitdown.py:324]: the tests seem repetitive. Could you wait until the test refactor PR is approved and merged and add tests for pathlib later?

**Review by l-lumin — CHANGES_REQUESTED** [review]

File: src/markitdown/_markitdown.py (hunk 1 above)
Comment (SigireddyBalasai) [reply src/markitdown/_markitdown.py:16]: i forget to remove it after trying to use different approach

File: tests/test_markitdown.py (hunk 3 above)
Comment (SigireddyBalasai) [reply tests/test_markitdown.py:324]: i need to wait for sometime before adding the tests right

File: src/markitdown/_markitdown.py (hunk 2 above)
Comment (SigireddyBalasai) [reply src/markitdown/_markitdown.py:1046]: done

**Comment by SigireddyBalasai** [comment]
@l-lumin please check the new changes

File: tests/test_markitdown.py [hunk 4]
```diff
@@ -7,6 +7,7 @@
 import requests
 
 from markitdown import MarkItDown
+from pathlib import Path
```
Comment (afourney) [inline tests/test_markitdown.py:10]: Are we using Path elsewhere in this tests file?

Otherwise, PR looks good.

**Review by l-lumin — APPROVED** [review]

File: tests/test_markitdown.py (hunk 4 above)
Comment (SigireddyBalasai) [reply tests/test_markitdown.py]: no usage of it

Comment (SigireddyBalasai) [reply tests/test_markitdown.py]: removed it thank you

**Review by SigireddyBalasai — COMMENTED** [review]
done the change

**Comment by SigireddyBalasai** [comment]
@afourney please let me know of status

**Review by gagb — APPROVED** [review]

File: src/markitdown/_markitdown.py [hunk 5]
```diff
@@ -1307,10 +1308,14 @@ def convert(
         # Request response
         elif isinstance(source, requests.Response):
             return self.convert_response(source, **kwargs)
+        elif isinstance(source, Path):
```
Comment (gagb) [inline src/markitdown/_markitdown.py:1311]: Is there a test for this?

**Review by gagb — APPROVED** [review]

**Pull request merged**